lib/posix/clock.c

```diff
@@ -74,6 +84,13 @@ int clock_settime(clockid_t clock_id, const struct timespec *tp)
 		return -1;
 	}
 
+#ifdef CONFIG_RTC
+	struct device *dev_rtc = device_get_binding(DT_RTC_0_NAME);
+	if (dev_rtc) {
+		return rtc_settime(dev_rtc, tp);
+	}
+#endif
+
```

Comment:
not an issue in your commit, but line 118 seems bogus. The ARG_UNUSED should be on tz, not tv.

Reply:
I made a pull request for that: https://github.com/zephyrproject-rtos/zephyr/pull/23550